google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AddAppEngineFrameworkSupportAction.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.facet;
+
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.intellij.facet.impl.DefaultFacetsProvider;
+import com.intellij.framework.addSupport.FrameworkSupportInModuleConfigurable;
+import com.intellij.framework.addSupport.FrameworkSupportInModuleProvider;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleManager;
+import com.intellij.openapi.module.ModuleType;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.ui.configuration.ChooseModulesDialog;
+import com.intellij.openapi.ui.Messages;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Action to add either App Engine Flexible or Standard framework support to a module which
+ * includes creating appropriate facet and run configurations.
+ */
+public abstract class AddAppEngineFrameworkSupportAction extends AnAction {
+  private final String FRAMEWORK_NAME_IN_MESSAGE;
+
+  public AddAppEngineFrameworkSupportAction(@NotNull String nameInTitle,
```

Comment:
package-private constructor?

Reply:
The classes that inherit are in different packages: com.google.cloud.tools.intellij.appengine.facet.standard and com.google.cloud.tools.intellij.appengine.facet.flexible

Reply:
sorry, missed that